terraform/sgtm-cluster/main.tf

```diff
@@ -0,0 +1,432 @@
+  default     = "main,master"
+}
+
+variable "sgtm_feature__check_rerun_on_approval_enabled" {
+  type        = string
+  description = "'true' if a check rerun should be rerequested when a PR is approved. Must be combined with sgtm_feature__check_rerun_base_ref_names and sgtm_feature__check_rerun_threshold_hours."
+  default     = "false"
+}
+
+variable "sgtm_rest_api_id" {
+  type        = string
+  description = "The ID of the API Gateway REST API for SGTM"
+}
+
+variable "sgtm_rest_api_root_resource_id" {
+  type        = string
+  description = "The ID of the root resource of the API Gateway REST API for SGTM"
+
+}
+
+variable "sgtm_rest_api_execution_arn" {
+  type        = string
+  description = "The ARN of the API Gateway execution role for SGTM"
+
+}
+
+locals {
+  suffix = var.naming_suffix != null ? "_${var.naming_suffix}" : ""
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+
+### LAMBDA
+
+resource "aws_lambda_function" "sgtm" {
+  s3_bucket        = var.lambda_code_s3_bucket_name
+  s3_key           = aws_s3_bucket_object.lambda_code_bundle.key
+  function_name    = "sgtm${local.suffix}"
+  role             = aws_iam_role.iam_for_lambda_function.arn
+  handler          = "src.handler.handler"
+  source_code_hash = data.archive_file.create_dist_pkg.output_base64sha256
+
+  runtime = var.lambda_runtime
+
+  timeout = var.lambda_function_timeout
+  environment {
+    variables = {
+      API_KEYS_S3_BUCKET                             = var.api_key_s3_bucket_name,
+      API_KEYS_S3_KEY                                = var.api_key_s3_object,
+      SGTM_FEATURE__AUTOMERGE_ENABLED                = var.sgtm_feature__automerge_enabled,
+      SGTM_FEATURE__AUTOCOMPLETE_ENABLED             = var.sgtm_feature__autocomplete_enabled,
+      SGTM_FEATURE__DISABLE_GITHUB_TEAM_SUBSCRIPTION = var.sgtm_feature__disable_github_team_subscription,
+      SGTM_FEATURE__ALLOW_PERSISTENT_TASK_ASSIGNEE   = var.sgtm_feature__allow_persistent_task_assignee,
+      SGTM_FEATURE__FOLLOWUP_REVIEW_GITHUB_USERS     = var.sgtm_feature__followup_review_github_users,
+      SGTM_FEATURE__CHECK_RERUN_THRESHOLD_HOURS      = var.sgtm_feature__check_rerun_threshold_hours,
+      SGTM_FEATURE__CHECK_RERUN_BASE_REF_NAMES       = var.sgtm_feature__check_rerun_base_ref_names,
+      SGTM_FEATURE__CHECK_RERUN_ON_APPROVAL_ENABLED  = var.sgtm_feature__check_rerun_on_approval_enabled
+      GITHUB_USERNAMES_TO_ASANA_GIDS_S3_PATH         = var.github_usernames_to_asana_gids_s3_path
+    }
+  }
+}
+
+resource "null_resource" "install_python_dependencies" {
+  triggers = {
+    src_sha1 = sha1(join("", [for f in fileset(path.root, "../src/**") : filesha1(f)]))
+  }
+
+  provisioner "local-exec" {
+    command = "bash ${path.module}/../scripts/create_pkg.sh"
+
+    environment = {
+      source_code_path = "../src"
+      function_name    = "sgtm"
```

Comment:
This might need to be changed to include the suffix, but I'm not 100% sure about that 🤔

Reply:
hmm in `scripts/create_pkg.sh` looks like it's only used to set up a virtual env so seems ok to leave this